src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -504,7 +516,17 @@ string GetDisplayText()
         }
     }
 
-    private void UpdateResourcesList() => _resources = GetConsoleLogResourceSelectViewModels(_resourceByName, _noSelection, Loc[nameof(Dashboard.Resources.ConsoleLogs.ConsoleLogsUnknownState)], _showHiddenResources);
+    private void UpdateResourcesList()
+    {
+        _resources = GetConsoleLogResourceSelectViewModels(_resourceByName, _noSelection, Loc[nameof(Dashboard.Resources.ConsoleLogs.ConsoleLogsUnknownState)], _showHiddenResources, out var optionToSelect);
+        if (optionToSelect is not null)
+        {
+            Debug.Assert(optionToSelect.Id?.InstanceId is not null);
+            PageViewModel.SelectedOption = optionToSelect;
+            PageViewModel.SelectedResource = _resourceByName[optionToSelect.Id.InstanceId];
+            NavigationManager.NavigateTo(GetUrlFromSerializableViewModel(ConvertViewModelToSerializable()));
```

Comment:
Is this required? The metrics page has already been changed to redirect when there is one app.